src/components/ViewSnippetHeader.tsx

```diff
@@ -62,6 +63,36 @@ export default function ViewSnippetHeader() {
       navigate("/editor?snippet_id=" + forkedSnippet.snippet_id)
     },
   })
+  const handleStarClick = async () => {
+    try {
+      if (isStarred) {
+        await axios.post("/snippets/remove_star", {
+          snippet_id: snippet!.snippet_id,
+        })
+        setIsStarred(false)
```

Comment:
use `qc.invalidateQueries(["snippets", snippet?.snippet_id])` (or whatever the query key is)